openadapt/start.py

```diff
@@ -0,0 +1,46 @@
+"""
+Implements the code needed to update the OpenAdapt app if needed.
+
+Usage:
+    python3 -m openadapt.start
+"""
+from loguru import logger
+import subprocess
+
+from openadapt.app.main import run_app
+
+
+def changes_needed(result: str) -> bool:
+    """Checks if git stash needs to be run.
+
+    Assumes result is the result of git status.
+    """
+    if "Changes to be committed:" in result:
+        return True
+    elif "Changes not staged for commit:" in result:
+        return True
+    return False
+
+
+def start_openadapt_app() -> None:
+    """
+    The main function which runs the OpenAdapt app when it is updated.
+    """
+    result = subprocess.run(["git", "status"], capture_output=True, text=True)
+
+    if changes_needed(result.stdout):
```

Comment:
Is this function needed? Git stash should do nothing if there is nothing to do, so I think we can just run `git stash` without checking `git status`. Or am I missing something?

Reply:
Addressed in [a87e202](https://github.com/MLDSAI/OpenAdapt/pull/201/commits/a87e20247926993821c147232d014f0cfd779e86)